influxql/engine.go

```diff
@@ -496,11 +496,19 @@ func (m *MapReduceJob) processAggregate(c *Call, reduceFunc ReduceFunc, resultVa
 		}
 	}
 
+	firstInterval := m.interval
+	if !m.stmt.IsRawQuery {
```

Comment:
Process aggregate gets called for ALL queries. With raw queries it's just that there is one aggregate and the call object is nil. See: https://github.com/influxdb/influxdb/blob/master/influxql/engine.go#L84-L89

The reason this passes with or without the check for `IsRaw` is because the interval value ends up being the same, I think. That's because the interval is set before this to be the time between tmax and tmin. This just does some math to end at the same result.

So yes, technically, we don't have to check that it's not a raw query.